src/test/java/io/stargate/sgv2/jsonapi/service/projection/DocumentProjectorTest.java

```diff
@@ -170,6 +171,87 @@ public void verifyNoDollarSimilarity() throws Exception {
     }
   }
 
+  // [jsonapi#1853]: Support ampersand-escape in projection
+  @Nested
+  class ProjectorPathValidation {
+    @Test
+    public void verifyAmpersandEscape() throws Exception {
+      final String docJson =
+          """
+                      "pricing": {
+                          "price.usd": 1.0,
+                          "price&jpy": 2.0,
+                          "price&.aud": 3.0,
+                          "app.kubernetes.io/name": {
+                              "abc$def": "123",
+                              "a b": "456"
+                          }
+                      }
+                      """;
+      final String projectionString =
+          """
+                      {
+                          "pricing.price&.usd": 1,
+                          "pricing.price&&jpy": 1,
+                          "pricing.price&&&.aud": 1,
```

Comment:
Might be better to have a test case where only some of paths are included -- otherwise test could work simply because the whole document was included?

Would also be good to have a nested entry without escaped characters.